hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTypeConverter.java

```diff
@@ -30,7 +37,52 @@ private HiveTypeConverter() {
 
   }
 
-  public static String convert(Type type) {
+  /**
+   * Converts the Iceberg schema to a Hive schema.
+   * @param schema The original Iceberg schema to convert
+   * @return The Hive column list generated from the Iceberg schema
+   */
+  public static List<FieldSchema> hiveSchema(Schema schema) {
```

Comment:
Most of the other modules use a different naming convention:

* `AvroSchemaUtil.convert` for Schema to Iceberg type and Iceberg type to Schema
* `ParquetSchemaUtil.convert`
* `SparkSchemaUtil.convert`
* `FlinkSchemaUtil.convert`

I think it would be nice to use the same convention here: use `HiveSchemaUtil` instead of `HiveTypeConverter` and then use overloads of `convert`. This would also make the API a bit less confusing because `HiveTypeConverter` and `HiveSchemaConverter` are very similar names and share a lot of the same tasks with different method args. Could these be combined into one?

Some schema classes have a different type for the top-level schema (Iceberg and Parquet) and we usually convert between the two. Here, I think you could convert between Iceberg `Schema` and `List<FieldSchema>`, and between Iceberg `Type` and `TypeInfo`.

Reply:
Thanks @rdblue for the review!

Renamed the class to HiveSchemaUtil and the methods to `convert`. Also added `convert` methods for Type conversion.
Kept the `HiveSchemaConverter` class for now for 2 reasons:
- We have to extend the actual converter for Hive3 (Timestamp with local timezone is a separate type in Hive 3)
- You suggested to implement a Hive type visitor on another review - we might want to use that for conversions when it is ready

Until we have a final solution I added a javadoc where specifically stated that this package private class should not be used anywhere for conversion and pointed to the correct class to use.

What do you think? Would this be good enough for start?

Thanks,
Peter

Reply:
Yes, sounds like you're heading in the direction I suggested. I'm flexible on how we get there and would defer to your judgement on it.